src/components/modal/addtostudio/container.jsx

```diff
@@ -17,7 +17,7 @@ class AddToStudioModal extends React.Component {
         };
     }
 
-    componentWillUpdate () {
+    componentDidUpdate () {
```

Comment:
Need to confirm this works

Reply:
Tested, works fine

Reply:
Seems risky to do all of these lifecycle updates in a lint PR. I agree we should update these so we can upgrade React, but maybe in a separate PR. Instead I think we should set our React version for `eslint-plugin-react`, like gui does: https://github.com/LLK/scratch-gui/blob/8eee001c12d0b40790da684c228bab1b32695100/src/.eslintrc.js#L22

Reply:
Hm, I tried this, but the setting doesn't seem to be being recognized by the linter.

I made the file look like:
```
module.exports = {
    extends: ['scratch', 'scratch/node'],
    plugins: ['json'],
    settings: {
        react: {
            version: '16.2' // Prevent 16.3 lifecycle method errors
        }
    }
};
```

...but when I run `npm run test:lint`, I see the warning:

```
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .
```

Reply:
Aah, I had to set it in `src/eslintrc.js` -- I had been setting it in the top-level `./eslintrc.js`